halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/profile/deck/DeckProfileFactory.java

```diff
@@ -86,6 +88,25 @@ protected void setProfile(
     bindings.put("timezone", deploymentConfiguration.getTimezone());
     bindings.put("version", deploymentConfiguration.getVersion());
 
+    KubernetesSettings ks = endpoints.getServiceSettings(Type.GATE).getKubernetes();
+
+    // TODO I think this is right, but need to check
+    Plugins plugins = deploymentConfiguration.getPlugins();
+    Map<String, String> pluginMetadata =
+        plugins.getPlugins().stream()
+            .filter(p -> p.getEnabled())
+            .filter(p -> !p.getManifestLocation().isEmpty())
+            .map(p -> Plugin.getUIAssets(p, p.generateManifest()))
+            .flatMap(m -> m.entrySet().stream())
+            .distinct()
+            .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+    ks.getVolumes()
+        .add(
+            new ConfigSource()
+                .setId("secretname")
+                .setMountPath("mountpath")
+                .setEnv(pluginMetadata));
```

Comment:
I was expecting this to create a secret with the pluginMetadata info, but it seems like it only added to Gate's container. I think this makes sense, the KubernetesSettings here refer to the kubernetes settings for Gate, but I'm not sure how to create secrets in the kubernetes cluster to be pulled from the init container

Reply:
For creating secrets I would look at this method: https://github.com/spinnaker/halyard/blob/7bfa5ce9315aaca90d62b2a17c533505da6e0bee/halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/service/distributed/kubernetes/v2/KubernetesV2Utils.java#L123 , since it seems that is used currently for creating the rest of the secrets for all services.